### DISCUSSION TITLE
How to find the right api to use for BRepBuilderAPI_MakeEdge

### DISCUSSION THREAD

**Discussion by BarbourSmith** [donalffons/opencascade.js#207] [Q&A]
Hello! Huge fan of this project! 

I am trying to get a handle on how the js interface to OpenCascade works. Specifically I'm looking at some of the code in Cascade Studio and trying to do something similar with the latest release of OpenCascade.js.

In the Cascade Studio source code they make a circle like this:

```
function Circle(radius, wire) {
  let curCircle = CacheOp(arguments, () => {
    let circle = new oc.GC_MakeCircle(new oc.gp_Ax2(new oc.gp_Pnt(0, 0, 0),
      new oc.gp_Dir(0, 0, 1)), radius).Value();
    let edge = new oc.BRepBuilderAPI_MakeEdge(circle).Edge();
    let circleWire = new oc.BRepBuilderAPI_MakeWire(edge).Wire();
    if (wire) { return circleWire; }
    return new oc.BRepBuilderAPI_MakeFace(circleWire).Face();
  });
  sceneShapes.push(curCircle);
  return curCircle;
}
```

But when I try to do something similar with the latest OpenCascade.js I'm having a hard time finding a version of `BRepBuilderAPI_MakeEdge` with the same interface in the docs here: https://ocjs.org/reference-docs/classes/BRepBuilderAPI_MakeEdge

So far I am doing:
```
let circle = new oc.GC_MakeCircle_2(new oc.gp_Ax2_3(new oc.gp_Pnt_3(0, 0, 0),
        new oc.gp_Dir_1()), radius).Value();
let edge = new oc.BRepBuilderAPI_MakeEdge_1(circle).Edge();
```

Any guidance for where I might be going wrong?

**Comment by sgenoud** [comment]
You can have a look at [how I do it in replicad here](https://github.com/sgenoud/replicad/blob/main/packages/replicad/src/shapeHelpers.ts#L27).

I use `BRepBuilderAPI_MakeEdge_8` (instead of `_1`). This is due to how opencascadejs handles [overloaded methods](https://ocjs.org/docs/advanced/differences-cpp-js/overloaded-methods).

**Comment by BarbourSmith** [comment]
I discovered replicad right after I asked and it looks perfect :)
